tasks/main.yml

```diff
@@ -223,15 +225,63 @@
         state: absent
       when: __mssql_curr_repo.stdout | length > 0
 
+- name: Prepare for the upgrade
+  when:
+    - __mssql_current_version is defined
+    - mssql_upgrade | bool
+    - ansible_pkg_mgr == "zypper"
+  block:
+    - name: >-
+        Update all packages from SQL Server repo
+        version {{ __mssql_current_version }}
+      package:
+        name: "*"
+        extra_args: >-
+          -r packages-microsoft-com-mssql-server-{{ __mssql_current_version }}
+        state: latest  # noqa package-latest
+
+    - name: Get zypper repositories  # noqa command-instead-of-module
+      shell: >-
+        set -euo pipefail;
+        zypper repos -E
+        | grep packages-microsoft-com-mssql-server-
+        | grep -v packages-microsoft-com-mssql-server-{{ mssql_version }}
+        | cut -d" " -f 4
+        || true
+      register: __mssql_curr_repo
```

Comment:
It's better to use a different name for the variable. e.g. `__mssql_zypper_curr_repo`. Just in case if we need to use the content of __mssql_curr_repo somewhere else, and the content is different for dnf and zypper.